code/modules/mod/mod_control.dm

```diff
@@ -688,8 +688,11 @@
 	wearer.update_spacesuit_hud_icon(state_to_use || "0")
 
 /obj/item/mod/control/proc/update_speed()
-	for(var/obj/item/part as anything in get_parts())
-		part.slowdown = slowdown_deployed / (length(mod_parts) - 1)
+	var/datum/mod_part/part_datum = get_part_datum(src)
+	for(var/obj/item/part as anything in get_parts(all = TRUE))
+		part.slowdown = slowdown_deployed / length(mod_parts)
+		if (!part_datum.sealed)
+			part.slowdown = max(part.slowdown, 0)
```

Comment:
use ``get_part_datum(part)`` instead of relying on MOD's own state

Reply:
When I tried this the first time it didnt work but you are right that my code is too naive in that you could power up the interdyne module and then deploy all the other parts and go faster even if theyre not sealed, not sure what the benefit to that is but it's not intended behaviour, I will dig deeper.